appservice/src/main/java/nu/yona/server/subscriptions/rest/UserController.java

```diff
@@ -582,7 +582,10 @@ public static UserResourceAssembler createInstanceForOwnUser(CurieProvider curie
 		public UserResource toResource(UserDto user)
 		{
 			UserResource userResource = instantiateResource(user);
-			addSelfLink(userResource);
+			if (representation != UserResourceRepresentation.MINIMAL || user.getPrivateData().isFetchable())
```

Comment:
You're reading this the wrong way. The link is added when the representation is not minimal or the private data is fetchable.

Reply:
Oops, I think so. But then, we do include when the representation is `BUDDY_USER` or `OWN_USER` even when the resource is not fetchable. So why not `representation != UserResourceRepresentation.MINIMAL && user.getPrivateData().isFetchable()`?

Reply:
I was hoping you had a logical explanation :)
To me, it also makes more sense to have an ``&&`` in the middle. I have no idea why I put an ``||`` there. I'll test it and see whether it can be changed.

Reply:
Thanks for pushing on! It turned out that the first part of the condition (``representation != UserResourceRepresentation.MINIMAL``) should be removed. It caused the inclusion of a self-link in all in user entities embedded in a buddy entity for which the buddy relationship wasn't fully established yet. I've corrected that now and updated the tests accordingly.